R/sample_pseudoabs.R

```diff
@@ -41,11 +41,13 @@
 
 sample_pseudoabs <- function (data, raster, n, coords = NULL,
                               method="random", class_label = "pseudoabs",
-                              return_pres=TRUE)
+                              return_pres = TRUE)
 {
   return_sf <- FALSE # flag whether we need to return an sf object
-  if (inherits(data,"sf")){
+  if (inherits(data,"sf")) {
+    if (!all(c("X", "Y") %in% names(data))) {
```

Comment:
Well spotted. But I think that it would make sense to go a step further, and make sure that `X` and `Y` are identical to what you would get from `st_coordinates(data)`. Could you please add that check if we do already have `X` and `Y`? In that way, we are sure that we are using the correct coordinates under all circumstances.